lib/gems_comparator/gem_info.rb

```diff
@@ -24,8 +24,11 @@ def compare_url
     end
 
     def github_url
-      urls = [homepage, source_code_uri, github_url_from_yaml]
-      urls.map(&method(:normalized_github_url)).compact.first
+      @github_url ||= [
+        source_code_uri,
+        homepage,
+        github_url_from_yaml
+      ].map(&method(:normalized_github_url)).compact.first
```

Comment:
In this way, when `first` returns `nil`, it cannot memorize the value.
So change it to check the instance variable in the guard clause.

```ruby
def github_url
  return @github_url if instance_variable_defined?(:@github_url)
```

Reply:
Exactly! I'm fixed this.